internal/apis/networking/validation/network.go

```diff
@@ -77,6 +101,35 @@ func validateNetworkPeering(peering networking.NetworkPeering, fldPath *field.Pa
 	return allErrs
 }
 
+func validatePeeringClaimRef(peeringClaimRef networking.NetworkPeeringClaimRef, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if len(peeringClaimRef.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), "name is required"))
+	} else {
+		for _, msg := range apivalidation.NameIsDNSLabel(peeringClaimRef.Name, false) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("name"), peeringClaimRef.Name, msg))
+		}
+	}
+
+	if peeringClaimRef.Namespace != "" {
+		for _, msg := range apivalidation.NameIsDNSLabel(peeringClaimRef.Namespace, false) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("namespace"), peeringClaimRef.Namespace, msg))
+		}
+	}
+
+	if peeringClaimRef.UID != "" && !isValidUID(string(peeringClaimRef.UID)) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("uid"), peeringClaimRef.UID, "invalid uid"))
+	}
+
+	return allErrs
+}
+
+func isValidUID(u string) bool {
```

Comment:
And then do something like this here:
```
func IsValidUUID(u string) bool {
    _, err := uuid.Parse(u)
    return err == nil
 }
```
We can also make this utility function public.